api/script/index.js

```diff
@@ -447,7 +447,8 @@ module.exports = {
                 }
                 return false;
             })
-            .then(finish, finish)
+            .then(finish)
+            .catch((error) => (this.error(error) && finish()))
```

Comment:
this.error() does not return anything, so `&&` will result in not calling finish()